core/node/consensus/src/en.rs

```diff
@@ -70,7 +71,11 @@ impl EN {
             s.spawn_bg::<()>(async {
                 let old = genesis;
                 loop {
-                    if let Ok(new) = self.fetch_genesis(ctx).await {
+                    if let Ok(new) = self
+                        .fetch_genesis(ctx)
+                        .instrument(tracing::info_span!("genesis_monitor_fetch"))
```

Comment:
Is there any specific rule for this naming (i.e. why not "fetch_genesis")?
Why here though? Shouldn't you rather add `#[tracing...]` to fetch_genesis method instead?
If you want to instrument the whole subtask, you can put it on the outer future, like `s.spawn_bg(async { ... }.instrument(info_span));`

Reply:
I also think if we want to add more tracing we should add them as an attribute to the method, it keeps the call site easier to read.

Reply:
Good point, this was a leftover from my attempt at covering more stuff here with spans. Changed to this being a method attribute